### PR TITLE
Show modal in same position (height) as trigger of modal

### LINKED ISSUE #72: Suggestions - Referreres popup get's centered in the middle of the list
If the list is of suggestions is long, the modal popup is almost not visible. In this case the popup is all the way at the bottom of the screen.

image

### PR DESCRIPTION
**Pull request by ellinge**
Fixes #72 

Had to do a bit of a workaround since the table / modal is within an iframe so not easily fixed by only css.

Either way the modal is now positioned based on the offset height of the button which triggered it.

image

### CODE REVIEW

**Comment by ellinge** [comment]
Not sure what to to with the build failing, tests seems to pass but I think your sonar-configuration is off.

**Comment by marisks** [comment]
The build fails for the reason you wrote. Please ignore it.
Regarding the changes, the fix should be possible without JS and just with CSS. Also, it would be good to update Bootstrap to the release version (now it uses Beta) before applying the fix - it might fix the issue. 
I will take a look at it.
If none of it will help, then this PR will be accepted.

**Comment by ellinge** [comment]
I don’t think it wont be doable with css because the iframe viewport.

**Comment by marisks** [comment]
You are right. Bootstrap already sets it in the center, but the issue is that it is in an iframe.

File: src/Geta.NotFoundHandler.Admin/wwwroot/GetaNotFoundHandlerAdmin/js/dashboard.js [hunk 1]
```diff
@@ -36,6 +35,25 @@
         });
     }
 
+    function adjustModalPosition() {
+        var modalTriggers = document.querySelectorAll('.modal-trigger');
+        modalTriggers.forEach(function (modalTrigger) {
+            modalTrigger.addEventListener('click',
+                function () {
```
Comment (marisks) [inline src/Geta.NotFoundHandler.Admin/wwwroot/GetaNotFoundHandlerAdmin/js/dashboard.js:42]: Please reduce nesting in this function. 
image

Checks for `modalTrigger.dataset.bsTarget` and `modalDialog` can return as it finds that an element does not exist.

**Review by marisks — CHANGES_REQUESTED** [review]

File: src/Geta.NotFoundHandler.Admin/wwwroot/GetaNotFoundHandlerAdmin/js/dashboard.js (hunk 1 above)
Comment (ellinge) [reply src/Geta.NotFoundHandler.Admin/wwwroot/GetaNotFoundHandlerAdmin/js/dashboard.js:42]: Should I update to closest release candidate to this beta of bootstrap as well still?

Comment (marisks) [reply src/Geta.NotFoundHandler.Admin/wwwroot/GetaNotFoundHandlerAdmin/js/dashboard.js:42]: I've updated it already.

**Comment by jevgenijsp** [comment]
The modal will go behind the screen If there are more than 10 referrers in the first entries.

Example: 
image


There should be a fix for such situations, or better just show it always centered taking into account the current scrolling position.

**Comment by jevgenijsp** [comment]
Closing this ticket due to creating a copy of this with the additional fix: https://github.com/Geta/geta-notfoundhandler/pull/147

**Pull request closed without merging**